examples/user_profiles/connector.py

```diff
@@ -0,0 +1,143 @@
+from fivetran_connector_sdk import Connector
+from fivetran_connector_sdk import Operations as op
+from fivetran_connector_sdk import Logging as log
+import requests as rq
+import pandas as pd
+
+
+def schema(configuration: dict):
+    return [
+        {
+            "table": "profile",
+            "primary_key": ["id"],
+            "columns": {
+                "id": "INT",
+                "gender": "STRING",
+                "fullName": "STRING",
+                "email": "STRING",
+                "age": "INT",
+                "date": "STRING",
+                "mobile": "STRING",
+                "nationality": "STRING"
+            }
+
+        },
+        {
+            "table": "location",
+            "primary_key": ["profileId"],
+            "columns": {
+                "profileId": "INT",
+                "street": "STRING",
+                "city": "STRING",
+                "state": "STRING",
+                "country": "STRING",
+                "postcode": "STRING"
+            }
+        },
+        {
+            "table": "Login",
+            "primary_key": ["profileId"],
+            "columns": {
+                "profileId": "INT",
+                "uuid": "STRING",
+                "username": "STRING",
+                "password": "STRING"
+            }
+        }
+    ]
+
+
+def update(configuration: dict, state: dict):
+    profile_cursor = state["profile_cursor"] if "profile_cursor" in state else 0
+
+    profile_df, location_df, login_df, cursor = get_data(profile_cursor)
+
+    # UPSERT all profile table data and checkpointing after every 5 records
+    for index, row in profile_df.iterrows():
+        yield op.upsert("profile", {col: row[col] for col in profile_df.columns})
+        if index % 5 == 0:
+            state["profile_cursor"] = row["id"]
+            yield op.checkpoint(state)
+    # Checkpointing at last
+    state["profile_cursor"] = cursor
+    yield op.checkpoint(state)
+
+    # Upserting data in "location" table in below loop
+    for index, row in location_df.iterrows():
+        yield op.upsert("location", {col: row[col] for col in location_df.columns})
```

Comment:
This is not a bad example, but would it not be better to have a version of upsert that acceps a dataframe so the customer doesn't have to write the looping code over and over?

Reply:
Having this as a minimum example is good while we decide if we are going to add a helper method or similar to accept a pd.DataFrame directly.

Reply:
I have added three different approaches to upsert the records from DataFrame. Please take a look.